x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/LogicalPlanBuilder.java

```diff
@@ -327,21 +322,18 @@ public PlanFactory visitRenameCommand(EsqlBaseParser.RenameCommandContext ctx) {
 
     @Override
     public PlanFactory visitKeepCommand(EsqlBaseParser.KeepCommandContext ctx) {
-        var identifiers = ctx.qualifiedNamePattern();
-        List<NamedExpression> projections = new ArrayList<>(identifiers.size());
-        boolean hasSeenStar = false;
-        for (QualifiedNamePatternContext patternContext : identifiers) {
-            NamedExpression ne = visitQualifiedNamePattern(patternContext);
+        final Holder<Boolean> hasSeenStar = new Holder<>(false);
```

Comment:
Oh interesting. In the rare cases I need this I tend to do `boolean[] hasSeenStar = new boolean[] {false};` - but this `Holder` dude is thing SQL long ago. It has a name which is nice.